docs/proposal/tls_cert.md

```diff
@@ -0,0 +1,140 @@
+## Certificate Application and Management Module
+
+### Summary
+
+kmesh supports TLS capabilities that require the use of certificates issued by istiod, so a set of certificate application and management modules is needed to apply for certificates from istiod and manage the lifecycle of the certificates.
+
+### Motivation
+
+kmesh needs to provide TLS capabilities for the managed workloads, and it needs to be able to conveniently apply for and manage certificates, adding, deleting, and refreshing certificates at the appropriate time.
+
+#### Goals
+
+1. Apply for certificates for the service accounts (sa) of the managed workloads
+2. Automatically renew certificates upon expiration
+
+#### Non-Goals
+
+1. In ambient mode, ztunnel and kmesh each have their own certificate management systems, and they do not interfere with each other, so it is possible that both apply for certificates for the same sa. In this case, the traffic will be handled by whichever system's certificate is in use.
+2. When kmesh restarts abnormally, all old certificate records are discarded, and all certificates are reapplied, without considering retaining the previous certificates.
+
+### Proposal
+
+Implement a certificate application module and a certificate management module, where:
+
+Certificate Application Module:
+- Establish an encrypted gRPC connection with istiod
+- Construct a CSR request and the corresponding private key for the service account (sa) of the managed workload
+- Use the CSR request to interact with istiod, and have istiod sign the certificate and return it
+
+Certificate Management Module:
+- Manage the timing of certificate operations: 1) When a new workload is added, 2) When a workload is deleted, 3) When a certificate is about to expire and needs to be automatically renewed
+- Manage the storage and management of certificates
+- Based on the certificate's expiration time, trigger the corresponding certificate refresh task when it's about to expire
+
+### Constraints
+
+Currently, if you want to use kmesh's TLS capabilities, you need to modify the deployment when starting istio, and add `kmesh-system/kmesh` to the CA_TRUSTED_NODE_ACCOUNTS environment variable.
+
+## Design Details
+
+### Certificate Application Module
+
+Create a `caclient` client when kmesh starts, and establish an encrypted gRPC connection with istiod.
+
+Use the information in the workload to construct a CSR request and a private key, send the CSR request to istiod via the `caclient`, and have istiod sign and return the certificate.
```

Comment:
what is `and have istiod sign` means? 
A long process can be illustrated in several steps